drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/util/SimulationSettingsCreationStrategy.java

```diff
@@ -15,46 +15,55 @@
  */
 package org.drools.workbench.screens.scenariosimulation.backend.server.util;
 
+import org.drools.scenariosimulation.api.model.AbstractScesimData;
+import org.drools.scenariosimulation.api.model.AbstractScesimModel;
+import org.drools.scenariosimulation.api.model.Background;
 import org.drools.scenariosimulation.api.model.ExpressionIdentifier;
 import org.drools.scenariosimulation.api.model.FactIdentifier;
 import org.drools.scenariosimulation.api.model.FactMapping;
 import org.drools.scenariosimulation.api.model.FactMappingType;
-import org.drools.scenariosimulation.api.model.Scenario;
-import org.drools.scenariosimulation.api.model.ScenarioWithIndex;
+import org.drools.scenariosimulation.api.model.ScesimDataWithIndex;
+import org.drools.scenariosimulation.api.model.ScesimModelDescriptor;
+import org.drools.scenariosimulation.api.model.Settings;
 import org.drools.scenariosimulation.api.model.Simulation;
-import org.drools.scenariosimulation.api.model.SimulationDescriptor;
 import org.uberfire.backend.vfs.Path;
 
 /**
- * <b>Strategy</b> that actually builds the required <code>Simulation</code> based on <code>ScenarioSimulationModel.Type</code>
+ * <b>Strategy</b> that actually builds the required <code>Simulation</code> and <code>Settings</code> based on <code>ScenarioSimulationModel.Type</code>
  */
-public interface SimulationCreationStrategy {
+public interface SimulationSettingsCreationStrategy {
 
     Simulation createSimulation(Path context, String value) throws Exception;
 
-    default ScenarioWithIndex createScenario(Simulation simulation, SimulationDescriptor simulationDescriptor) {
+    Background createBackground(Path context, String dmnFilePath) throws Exception;
```

Comment:
Since this interface is used not only by dmn, but rules, I propose  to change the dmnFilePath parameter.